src/Components/Components/ref/Microsoft.AspNetCore.Components.netstandard2.0.cs

```diff
@@ -676,55 +664,19 @@ public sealed partial class FieldChangedEventArgs
-        protected string CurrentValueAsString { get { throw null; } set { } }
-        protected Microsoft.AspNetCore.Components.Forms.EditContext EditContext { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } }
-        protected string FieldClass { get { throw null; } }
-        protected Microsoft.AspNetCore.Components.Forms.FieldIdentifier FieldIdentifier { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } }
-        [Microsoft.AspNetCore.Components.ParameterAttribute]
-        protected string Id { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } }
-        protected virtual string FormatValueAsString(T value) { throw null; }
-        public override System.Threading.Tasks.Task SetParametersAsync(Microsoft.AspNetCore.Components.ParameterCollection parameters) { throw null; }
-        protected abstract bool TryParseValueFromString(string value, out T result, out string validationErrorMessage);
-    }
-    public partial class InputCheckbox : Microsoft.AspNetCore.Components.Forms.InputBase<bool>
+    public partial class InputCheckbox : Microsoft.AspNetCore.Components.ComponentBase
     {
         public InputCheckbox() { }
         protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder) { }
         protected override bool TryParseValueFromString(string value, out bool result, out string validationErrorMessage) { throw null; }
     }
-    public partial class InputDate<T> : Microsoft.AspNetCore.Components.Forms.InputBase<T>
-    {
-        public InputDate() { }
-        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder) { }
-        protected override string FormatValueAsString(T value) { throw null; }
-        protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage) { throw null; }
-    }
-    public partial class InputNumber<T> : Microsoft.AspNetCore.Components.Forms.InputBase<T>
-    {
-        public InputNumber() { }
-        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder) { }
-        protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage) { throw null; }
-    }
-    public partial class InputSelect<T> : Microsoft.AspNetCore.Components.Forms.InputBase<T>
-    {
-        public InputSelect() { }
-        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder) { }
-        protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage) { throw null; }
-    }
-    public partial class InputText : Microsoft.AspNetCore.Components.Forms.InputBase<string>
+    public partial class InputText : Microsoft.AspNetCore.Components.ComponentBase
```

Comment:
Sorry to still be asking questions, but is it really OK to declare `InputText` as inheriting directly from `ComponentBase`, when it really inherits from `InputBase<string>`?

Is there any risk that this will make the tooling think there are no `Id` or `Class` parameters for `InputText`?

Reply:
whoa, I'm glad you caught this. This looks like a bug in the ref assembly generation, and is not intended.

Reply:
I will move these types to be manual as well and report this issue.

Reply:
https://github.com/dotnet/arcade/issues/2345